contracts/margin/external/exchangewrappers/ZeroExV2MultiOrderExchangeWrapper.sol

```diff
@@ -205,30 +251,78 @@ contract ZeroExV2MultiOrderExchangeWrapper is
                 );
             }
 
+            // ignore orders that the maker will not be able to fill
+            if (!makerHasEnoughTokens(
+                makerToken,
+                balances,
+                order.makerAddress,
+                available.makerAmount)
+            ) {
+                continue;
+            }
+
             // update the running tallies
             total.takerAmount = total.takerAmount.add(available.takerAmount);
             total.makerAmount = total.makerAmount.sub(available.makerAmount);
         }
+    }
 
-        // require that all amount was bought
-        require(
-            total.makerAmount == 0,
-            "ZeroExV2MultiOrderExchangeWrapper#getExchangeCost: Cannot buy enough maker token"
-        );
+    /**
+     * Checks and modifies balances to keep track of the expected balance of the maker after filling
+     * each order. Returns true if the maker has enough makerToken left to transfer amount.
+     */
+    function makerHasEnoughTokens(
+        address makerToken,
+        TokenBalance[] memory balances,
+        address makerAddress,
+        uint256 amount
+    )
+        private
+        view
+        returns (bool)
+    {
+        // find the maker's balance in the cache or the first non-populated balance in the cache
+        TokenBalance memory current;
+        uint256 i;
+        for (i = 0; i < balances.length; i++) {
+            current = balances[i];
+            if (
+                current.owner == address(0)
```

Comment:
I was trying to think if someone making a 0x order with maker address 0 could break this. I guess you'd never be able to sign such an order validly, however you could still call `getExchangeCost`. I guess it basically wouldn't do any good though since `getExchangeCost` is a constant function

Reply:
idk maybe just revert if `makerAddress == 0` tho

Reply:
We will never send in orders with a makerAddress of zero, doesn't need to be in the contract

Reply:
Much like our other 0x exchange wrappers, the exchangewrapper doesn't actually check for valid signatures